arrows/vxl/tests/test_high_pass_filter.cxx

```diff
@@ -0,0 +1,152 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/**
+ * \file
+ * \brief test VXL high pass filtering
+ */
+
+#include <test_gtest.h>
+
+#include <arrows/vxl/high_pass_filter.h>
+#include <arrows/vxl/image_container.h>
+#include <arrows/vxl/image_io.h>
+
+#include <vital/plugin_loader/plugin_manager.h>
+
+#include <gtest/gtest.h>
+
+namespace kv = kwiver::vital;
+namespace ka = kwiver::arrows;
+
+kv::path_t g_data_dir;
+static std::string test_image_name = "images/small_grey_logo.png";
+static std::string test_color_image_name = "images/small_color_logo.png";
+static std::string expected_box = "images/box.png";
+static std::string expected_box_wide = "images/box_wide.png";
+static std::string expected_bidir = "images/bidir.png";
+static std::string expected_bidir_wide = "images/bidir_wide.png";
+static std::string expected_bidir_color = "images/bidir_color.png";
+
+// ----------------------------------------------------------------------------
+int
+main( int argc, char** argv )
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  TEST_LOAD_PLUGINS();
+
+  GET_ARG( 1, g_data_dir );
+
+  return RUN_ALL_TESTS();
+}
+
+// ----------------------------------------------------------------------------
+class high_pass_filter : public ::testing::Test
+{
+  TEST_ARG( data_dir );
+};
+
```

Comment:
Are there any failure modes that need to be tested in addition to the various filter modes?

Reply:
I couldn't think of anything terribly important but I'd welcome suggestions if you have any. As an aside, unit tests on a number of these related PRs could be simplified but I'd prefer to go back and update them as a batch on a separate PR since I think they're fine now.